### PR TITLE
fix(policies): columns not always showing correctly

### PR DESCRIPTION
**Pull request by kleinfreund**
fix(policies): add regression test for columns not always showing correctly

fix(policies): columns not always showing correctly

Trigger a re-render of AppCollection when the current policy type changes to make sure the table re-renders with the correct columns.

refactor(policies): move route access into list view

Signed-off-by: Philipp Rudloff <philipp.rudloff@konghq.com>

### CODE REVIEW

File: features/mesh/policies/Index.feature [hunk 1]
```diff
@@ -147,3 +147,12 @@ Feature: mesh / policies / index
     And the "[data-testid='policy-type-link-MeshFaultInjection']" element exists
     # Always shows MeshGateway
     And the "[data-testid='policy-type-link-MeshGateway']" element exists
+
+  Scenario: Regression test: Zone column is visible when navigating from legacy policy type
```
Comment (johncowen) [inline features/mesh/policies/Index.feature:151]: Aren't all tests regression tests? i.e. bit of a nit, but do we need this prefix?

**Review by johncowen — COMMENTED** [review]
I had 1 little nitty question, but I was wondering in which PR this regressed?

File: features/mesh/policies/Index.feature (hunk 1 above)
Comment (kleinfreund) [reply features/mesh/policies/Index.feature:151]: I think it’s useful to know when a test is dedicated to a regression.

**Comment by kleinfreund** [comment]
> I had 1 little nitty question, but I was wondering in which PR this regressed?

https://github.com/kumahq/kuma-gui/pull/2119

**Comment by johncowen** [comment]
Oh was it the removal of the key here then?

https://github.com/kumahq/kuma-gui/pull/2119/files#diff-15b4ba78b79c8245ad6c3be64f3cf17c428c9de4cb80d83cb54f1382786ebf14L55

**Comment by kleinfreund** [comment]
> Oh was it the removal of the key here then?
> 
> https://github.com/kumahq/kuma-gui/pull/2119/files#diff-15b4ba78b79c8245ad6c3be64f3cf17c428c9de4cb80d83cb54f1382786ebf14L55

Since that is the only code change there, yes. I could’ve also re-introduced the `key` on PolicyList, but figured this was working as well.

**Comment by johncowen** [comment]
How possible would it be to split this `PolicyList` component into two nested routes/`*View.vue`s?

- `/gui/meshes/default/policies` <-- this would be the "policy type list" that is currently on the left
  - `/gui/meshes/default/policies/:type` <-- this would be the "policy list" that is currently on the right

If I follow correctly what the issue is, this would mean that every time you click a "policy type" on the left, the "policy" table on the right would completely re-render just through building this with nested routes rather than using things like `:key`.

**Comment by kleinfreund** [comment]
> How possible would it be to split this `PolicyList` component into two nested routes/`*View.vue`s?
> 
> * `/gui/meshes/default/policies` <-- this would be the "policy type list" that is currently on the left
>   
>   * `/gui/meshes/default/policies/:type` <-- this would be the "policy list" that is currently on the right
> 
> If I follow correctly what the issue is, this would mean that every time you click a "policy type" on the left, the "policy" table on the right would completely re-render just through building this with nested routes rather than using things like `:key`.

Sure, that’s possible. It’s also much, much, much more effort than just putting the `key` attribute here.

**Review by johncowen — APPROVED** [review]
👍 I made an issue to change this to use nested routes

https://github.com/kumahq/kuma-gui/issues/2162

**Pull request merged**